dev/NavigationView/NavigationView.cpp

```diff
@@ -194,6 +220,15 @@ void NavigationView::OnSelectionModelChildrenRequested(const winrt::SelectionMod
     {
         e.Children(children);
     }
+    else
+    {
+        e.Children(e.Source());
```

Comment:
If i am understanding this correctly, this seems to be only required when resolving the children for the first level of the indexpath (footer vs non-footer items).
I think it might make more sense to put this in an explicit if statement at the beginning of the if/else chain since we know exactly the scenario we want this to be executed (resolving children for the very first level).

Reply:
We now have no default behavior, is that okay?

---
In reply to: [463346168](https://github.com/microsoft/microsoft-ui-xaml/pull/1997#discussion_r463346168) [](ancestors = 463346168)

Reply:
The default scenario here would be hit if NavigationView can't resolve the items for a specific indexpath. Generally this is a scenario we should not be hitting as IndexPaths are constructed internally only and if we hit this scenario, something has seriously gone wrong.